src/Events.ts

```diff
@@ -46,18 +46,23 @@ export function omitEvents<
     }, {} as any)
 }
 
-export function createEventBus<C extends EventDeclaration>(args: {
-    events: C
-    channels?: Channel[]
-    ignoredEvents?: Array<keyof C>
-}): C {
+export function createEventBus<
+    C extends EventDeclaration,
+    T extends Array<keyof C>
+>(args: {
+    events: C;
+    channels?: Channel[];
+    ignoredEvents?: T;
+}): Omit<C, T[number]> {
 
     const transports = (args.channels || []).map(
         (c) => new Transport(c, (args.ignoredEvents as string[]))
     )
 
     const eventBus: Partial<C> = {}
-    for (const event in args.events) {
+    const filteredEvent = omitEvents(args.events, args.ignoredEvents ?? []) as Partial<C>
```

Comment:
Why this rather than simply skipper the `ignoredEvents` in the `for` loop ?
Also the `omitEvents` function can remain local and not be exported I believe.

Reply:
That's actually a good point! I filtered directly in the if statement, removed the omitEvent as it's not used anymore (removed the corresponding test) and added a missing test to cover the filtered_event case